app/src/main/java/org/schabi/newpipe/player/VideoPlayer.java

```diff
@@ -285,6 +288,12 @@ private void buildCaptionMenu(final List<String> availableLanguages) {
         if (captionPopupMenu == null) return;
         captionPopupMenu.getMenu().removeGroup(captionPopupMenuGroupId);
 
+        String userPreferredLanguage = PreferenceManager.getDefaultSharedPreferences(context)
+                .getString(context.getString(R.string.caption_user_set_key), null);
+        // english (auto-generated)
+        boolean searchForAutogenerated = userPreferredLanguage == null ||
+                !userPreferredLanguage.contains("(");
```

Comment:
What does `!userPreferredLanguage.contains("(")` do? Please add a comment to ensure people can understand your code later.

Reply:
if the user already has set the auto-generated language (hence the check for "("), it is not useful to look for that as the first part of the if condition would match the caption language directly.

Reply:
Thanks